flight/Modules/OnScreenDisplay/osd_menu.c

```diff
@@ -1671,6 +1816,195 @@ void sticklimits_menu(void)
+				}
+			}
+		}
+
+		uint8_t *target;
+		uint8_t max_val;
+		const uint16_t *band_ptr;
+		switch(settings.Band) {
+			case VTXSETTINGS_BAND_BAND5G8A:
+				band_ptr = BAND_5G8_A_FREQS;
+				max_val = VTXSETTINGS_BAND_5G8_A_FREQUENCY_MAXOPTVAL;
+				target = &settings.Band_5G8_A_Frequency;
+				break;
+			case VTXSETTINGS_BAND_BAND5G8B:
+				band_ptr = BAND_5G8_B_FREQS;
+				max_val = VTXSETTINGS_BAND_5G8_B_FREQUENCY_MAXOPTVAL;
+				target = &settings.Band_5G8_B_Frequency;
+				break;
+			case VTXSETTINGS_BAND_BAND5G8E:
+				band_ptr = BAND_5G8_E_FREQS;
+				max_val = VTXSETTINGS_BAND_5G8_E_FREQUENCY_MAXOPTVAL;
+				target = &settings.Band_5G8_E_Frequency;
+				break;
+			case VTXSETTINGS_BAND_AIRWAVE:
+				band_ptr = AIRWAVE_FREQS;
+				max_val = VTXSETTINGS_AIRWAVE_FREQUENCY_MAXOPTVAL;
+				target = &settings.Airwave_Frequency;
+				break;
+			default:
+			case VTXSETTINGS_BAND_RACEBAND:
+				band_ptr = RACEBAND_FREQS;
+				max_val = VTXSETTINGS_RACEBAND_FREQUENCY_MAXOPTVAL;
+				target = &settings.Raceband_Frequency;
+				break;
+		}
+		y_pos += MENU_LINE_SPACING;
+
+		sprintf(tmp_str, "    Channel: CH %d %d MHz", *target + 1, band_ptr[*target]);
+		write_string(tmp_str, MENU_LINE_X, y_pos, 0, 0, TEXT_VA_TOP, TEXT_HA_LEFT, 0, MENU_FONT);
+
+		if (current_state == FSM_STATE_VTX_CH) {
+			draw_selected_icon(MENU_LINE_X - 4, y_pos + 4);
+
+			if (current_event == FSM_EVENT_RIGHT) {
+				if (*target == max_val) {
+					*target = 0;
+				}
+				else {
+					*target += 1;
+				}
+			}
+			if (current_event == FSM_EVENT_LEFT) {
+				if (*target == 0) {
+					*target = max_val;
+				}
+				else {
+					*target -= 1;
+				}
+			}
+		}
+
+		y_pos += MENU_LINE_SPACING;
+		sprintf(tmp_str, "    Power:   %d mW", VTX_POWER[settings.Power]);
+		write_string(tmp_str, MENU_LINE_X, y_pos, 0, 0, TEXT_VA_TOP, TEXT_HA_LEFT, 0, MENU_FONT);
+
+		if (current_state == FSM_STATE_VTX_POWER) {
+			draw_selected_icon(MENU_LINE_X - 4, y_pos + 4);
+
+			if (current_event == FSM_EVENT_RIGHT) {
+				if (settings.Power == VTXSETTINGS_POWER_MAXOPTVAL) {
+					settings.Power = 0;
+				}
+				else {
+					settings.Power += 1;
+				}
+			}
+			if (current_event == FSM_EVENT_LEFT) {
+				if (settings.Power == 0) {
+					settings.Power = VTXSETTINGS_POWER_MAXOPTVAL;
+				}
+				else {
+					settings.Power -= 1;
+				}
+			}
+		}
+	}
+
+	y_pos += 2 * MENU_LINE_SPACING;
+	write_string("Apply", MENU_LINE_X, y_pos, 0, 0, TEXT_VA_TOP, TEXT_HA_LEFT, 0, MENU_FONT);
+	if (current_state == FSM_STATE_VTX_APPLY) {
+		draw_selected_icon(MENU_LINE_X - 4, y_pos + 4);
+		if (current_event == FSM_EVENT_RIGHT) {
+			VTXSettingsSet(&settings);
+		}
+	}
+
+	y_pos += MENU_LINE_SPACING;
+	write_string("Save and Exit", MENU_LINE_X, y_pos, 0, 0, TEXT_VA_TOP, TEXT_HA_LEFT, 0, MENU_FONT);
+	if (current_state == FSM_STATE_VTX_SAVEEXIT) {
+		draw_selected_icon(MENU_LINE_X - 4, y_pos + 4);
+		if (current_event == FSM_EVENT_RIGHT) {
+			VTXSettingsSet(&settings);
+			UAVObjSave(VTXSettingsHandle(), 0);
```

Comment:
do we need to `draw_selected_icon(MENU_LINE_X - 4, y_pos + 4);` here to re-draw (exit)?